### PR TITLE
fix: handle single files in copy tasks

### PR DESCRIPTION
**Pull request by Doggermelon**
This should fix FTB imported instances not having any mods installed.

Signed-off-by: Kemonomodoki <kemonomodoki@firemail.cc>

### CODE REVIEW

File: launcher/FileSystem.cpp [hunk 1]
```diff
@@ -174,36 +176,57 @@ bool copy::operator()(const QString& offset)
...
 
     fs::copy_options opt = copy_opts::none;
 
     // The default behavior is to follow symlinks
     if (!m_followSymlinks)
         opt |= copy_opts::copy_symlinks;
 
+    const auto testAndCopy = [opt, &err](const QString& s, const QString& d) {
+        if (ensureFilePathExists(d)) {
+            fs::copy(toStdString(s), toStdString(d), opt, err);
```
Comment (ghost) [inline launcher/FileSystem.cpp:189]: I'm pretty sure this can be done cleaner with Qt

Comment (Doggermelon) [reply launcher/FileSystem.cpp:189]: I decided to keep this as the function was changed to use std's filesystem 5932f36285ebf3ec7d852db1b80630589224eccc. I'm not too familiar with Qt, so I wouldn't know if there's an idiomatic way for this, but I can change it to Qt only if that's what you want.

Comment (ghost) [reply launcher/FileSystem.cpp:189]: I'm not the final authority anymore, but I trust what worked for (presumably) years more than what some random dude changed.

Comment (ghost) [reply launcher/FileSystem.cpp:189]: Also; I can help you with converting to Qt if that's what we plan to do

Comment (LennyMcLennington) [reply launcher/FileSystem.cpp:189]: > I'm not the final authority anymore, but I trust what worked for (presumably) years more than what some random dude changed.

fs::copy was what was used for years though? So you trust fs::copy then? Or are you referencing something else?

Comment (ghost) [reply launcher/FileSystem.cpp:189]: I'm not specifically referencing fs::copy, I'm referencing what flowln changed.

File: launcher/FileSystem.cpp [hunk 2]
```diff
@@ -174,36 +176,57 @@ bool copy::operator()(const QString& offset)
     auto src = PathCombine(m_src.absolutePath(), offset);
     auto dst = PathCombine(m_dst.absolutePath(), offset);
 
-    std::error_code err;
+    std::error_code err{};
```
Comment (LennyMcLennington) [inline launcher/FileSystem.cpp:179]: This variable is never used outside testAndCopy. Couldn't it be local to testAndCopy?

**Review by LennyMcLennington — CHANGES_REQUESTED** [review]

**Review by LennyMcLennington — APPROVED** [review]

**Pull request merged**